Sources/MapboxNavigation/OrnamentsController.swift

```diff
@@ -70,12 +72,41 @@ extension NavigationMapView {
                                                    selector: #selector(orientationDidChange(_:)),
                                                    name: UIDevice.orientationDidChangeNotification,
                                                    object: nil)
+            NotificationCenter.default.addObserver(self,
+                                                   selector: #selector(didUpdateEHorizonRoadName),
+                                                   name: .electronicHorizonDidUpdatePosition,
+                                                   object: nil)
+
         }
         
         private func suspendNotifications() {
             NotificationCenter.default.removeObserver(self,
                                                       name: UIDevice.orientationDidChangeNotification,
                                                       object: nil)
+            NotificationCenter.default.removeObserver(self,
+                                                      name: .electronicHorizonDidUpdatePosition,
+                                                      object: nil)
+        }
+        
+        @objc func didUpdateEHorizonRoadName(_ notification: Notification) {
+            guard let startingEdge = notification.userInfo?[RoadGraph.NotificationUserInfoKey.treeKey] as? RoadGraph.Edge else { return }
+            let edgeIdentifier = startingEdge.identifier
+            
+            guard let routeController = navigationViewData.router as? RouteController,
+                  let metadata = routeController.roadGraph.edgeMetadata(edgeIdentifier: edgeIdentifier) else { return }
+            
+            electronicHorizonRoadNames = metadata.names.map { name -> String in
+                switch name {
+                case .name(let name):
+                    return name
+                case .code(let code):
+                    return "(\(code))"
```

Comment:
A code is unlikely to match a queried feature’s name, but at least it would be slightly more likely without the parentheses. We would want to show the user a shield rather than a raw route number, but if we had to show a raw route number, it wouldn’t need to be in parentheses.

Reply:
Thanks for the suggestion, it would return the code directly.